### PR TITLE
Receive font size from android app and assign a listener for updates

### PR DESCRIPTION
**Pull request by tuncaulubilge**
Bridges the initial font size and adds listeners for continuous updates. Font size is passed into articles through the context package as the scale.

### CODE REVIEW

File: android-app/pages/article.js [hunk 1]
```diff
@@ -31,26 +32,52 @@ const platformAdConfig = {
...
-      onVideoPress={onVideoPress}
-      onTopicPress={onTopicPress}
-      platformAdConfig={adConfig}
-    />
-  );
-};
+class ArticleView extends Component {
+  constructor(props) {
+    super(props);
+    this.subscription = null;
```
Comment (craigbilner) [inline android-app/pages/article.js]: is there value in this?

File: android-app/pages/article.js [hunk 2]
```diff
@@ -31,26 +32,51 @@ const platformAdConfig = {
...
-      onVideoPress={onVideoPress}
-      onTopicPress={onTopicPress}
-      platformAdConfig={adConfig}
-    />
-  );
-};
+class ArticleView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { fontSize: this.props.fontSize };
```
Comment (colincclark) [inline android-app/pages/article.js]: please destruct `fontSize` above this from `props`

File: android-app/pages/article.js [hunk 3]
```diff
@@ -31,26 +32,51 @@ const platformAdConfig = {
...
+
+  render() {
+    const { articleId, sectionName } = this.props;
+    const adConfig = { ...platformAdConfig, sectionName };
+
+    return (
+      <ArticlePageView
+        articleId={articleId}
+        analyticsStream={track}
+        fontSize={this.state.fontSize}
```
Comment (colincclark) [inline android-app/pages/article.js]: neater to destruct above

File: packages/pages/src/article.js [hunk 4]
```diff
@@ -54,7 +56,12 @@ ArticleDetailsPage.propTypes = {
   onAuthorPress: PropTypes.func.isRequired,
   onVideoPress: PropTypes.func.isRequired,
   onLinkPress: PropTypes.func.isRequired,
-  onTopicPress: PropTypes.func.isRequired
+  onTopicPress: PropTypes.func.isRequired,
+  fontSize: PropTypes.string
```
Comment (colincclark) [inline packages/pages/src/article.js]: easier to read if these remain alphabetised (optional)

Comment (tuncaulubilge) [reply packages/pages/src/article.js]: Didn't we have a lint rule for this? Does it only work for props but not for prop-types?

Comment (colincclark) [reply packages/pages/src/article.js]: Correct. Only for props atm. We are intorducing these rules gently, lol

File: android-app/pages/article.js [hunk 5]
```diff
@@ -31,26 +32,53 @@ const platformAdConfig = {
...
-};
+class ArticleView extends Component {
+  constructor(props) {
+    super(props);
+    const { fontSize } = this.props;
+    this.state = { fontSize };
+  }
+
+  componentDidMount() {
+    this.subscription = articleEventEmitter.addListener(
```
Comment (craigbilner) [inline android-app/pages/article.js]: @tuncaulubilge did you say this can go now?

Comment (tuncaulubilge) [reply android-app/pages/article.js]: If we remove the button from the toolbar, we can also remove these. Let's keep it in for now as it'll make dev easier

File: android-app/pages/article.js [hunk 6]
```diff
@@ -31,26 +33,53 @@ const platformAdConfig = {
...
-      onVideoPress={onVideoPress}
-      onTopicPress={onTopicPress}
-      platformAdConfig={adConfig}
-    />
-  );
-};
+class ArticleView extends Component {
+  constructor(props) {
+    super(props);
+    const { scale } = this.props;
```
Comment (colincclark) [inline android-app/pages/article.js]: can just do `const { scale } = props;` here

**Comment by craigbilner** [comment]
will raise a different one

**Pull request closed without merging**